File: y15/d01/golang/main.go
package main

import (
	"flag"
	"fmt"
	"strings"
)

func main() {
	f := flag.String("f", "", "Input File Name")
	flag.Parse()
	lines := readLines(*f)
	solve(lines)
}

func solve(lines []string) {
	input := strings.Split(lines[0], "")
	floor := 0
	pos := 0
	for i, p := range input {
		// part 1
		if p == "(" {
			floor++
		} else {
			floor--
		}
		// part 2
		if floor == -1 && pos == 0 {
			pos = i + 1
		}

	}

	fmt.Println("Part 1", floor)
	fmt.Println("Part 2", pos)
}

File: y15/d01/rust/src/main.rs
mod helper;

use helper::read_lines;
use std::env;

fn main() {
    let args: Vec<String> = env::args().collect();
    if args.len() > 1 {
        let file = &args[1];
        println!("File: {}", file);
        let lines = read_lines(file);
        solve(lines);
    } else {
        println!("No file name provided");
    }
}

fn solve(lines: Vec<String>) {
    let input: Vec<char> = lines[0].chars().collect();

    let mut floor = 0;
    let mut basement_at = -1;
    for (i, c) in input.iter().enumerate() {
        if *c == '(' {
            floor += 1
        } else {
            floor -= 1
        }

        if basement_at == -1 && floor == -1 {
            basement_at = i as i32 + 1
        }
    }
    println!("{}", floor);
    println!("{}", basement_at);
}
